File: mobileweb/link.php
<?php
session_start();
if (!isset($_SESSION['username'])) {
    header("Location: index.php");
    exit();
}

require_once '../config/database.php';

$db = new Database();
$conn = $db->getConnection();

$username = $_SESSION['username'];

// Check if car is already linked and user has battery level
$isCarLinked = false;
$hasBatteryData = false;

if ($conn) {
    // Check if user has battery data (equivalent to car being linked)
    $stmt = $conn->prepare("SELECT COUNT(*) FROM battery_levels WHERE username = :username");
    $stmt->bindParam(':username', $username);
    $stmt->execute();
    $hasBatteryData = $stmt->fetchColumn() > 0;

    if ($hasBatteryData) {
        // Get battery level
        $stmt = $conn->prepare("SELECT battery_level FROM battery_levels WHERE username = :username");
        $stmt->bindParam(':username', $username);
        $stmt->execute();
        $result = $stmt->fetch(PDO::FETCH_ASSOC);
        $batteryLevel = (int)($result['battery_level'] ?? 0);

        if ($batteryLevel != -1) {
            // Car is linked and battery is initialized - redirect to dashboard
            header("Location: dashboard.php");
            exit();
        }
    }
}

// Handle form submissions
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    if (isset($_POST['link_car'])) {
        // Link car functionality
        if ($conn && !$hasBatteryData) {
            // Initialize battery to random level (15-50%) only if not already initialized
            $batteryLevel = 15 + rand(0, 35); // 15 to 50

            $stmt = $conn->prepare("INSERT INTO battery_levels (username, battery_level, initial_battery_level, battery_capacity_kwh) VALUES (:username, :battery_level, :initial_battery_level, 40.0)");
            $stmt->bindParam(':username', $username);
            $stmt->bindParam(':battery_level', $batteryLevel);
            $stmt->bindParam(':initial_battery_level', $batteryLevel);

            if ($stmt->execute()) {
                // Set car as linked in session
                $_SESSION['car_linked'] = true;

                // Redirect to dashboard
                header("Location: dashboard.php");
                exit();
            } else {
                $error = "Failed to link car. Please try again.";
            }
        }
    }
}
?>
<!DOCTYPE HTML>
<html>
<head>
    <title>Link Your Car - Cephra</title>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
    <link rel="icon" type="image/png" href="images/logo.png" />
    <link rel="stylesheet" href="assets/css/main.css" />
    <style>
        .link-container {
            text-align: center;
            padding: 20px;
        }
        .link-button {
            background: #4CAF50;
            color: white;
            padding: 15px 30px;
            border: none;
            border-radius: 5px;
            font-size: 16px;
            cursor: pointer;
            margin: 20px 0;
        }
        .link-button:disabled {
            background: #cccccc;
            cursor: not-allowed;
        }
        .terms-checkbox {
            margin: 20px 0;
            font-size: 14px;
        }
        .nav-buttons {
            position: fixed;
            bottom: 20px;
            left: 50%;
            transform: translateX(-50%);
            display: flex;
            gap: 20px;
        }
        .nav-button {
            width: 50px;
            height: 50px;
            border-radius: 50%;
            border: none;
            background: #007bff;
            color: white;
            cursor: pointer;
            display: flex;
            align-items: center;
            justify-content: center;
            font-size: 20px;
        }
        .error-message {
            color: red;
            margin: 10px 0;
        }
    </style>
</head>
<body class="homepage is-preload">
    <div id="page-wrapper">
        <!-- Header -->
        <div id="header-wrapper">
            <!-- Header -->
            <header id="header">
                <div class="inner">
                    <!-- Logo -->
                    <h1>
                        <a href="dashboard.php" id="logo">Cephra</a>
                    </h1>
                    <!-- Nav -->
                    <nav id="nav">
                        <ul>
                            <li><a href="dashboard.php">Home</a></li>
                            <li class="current_page_item"><a href="link.php">Link</a></li>
                            <li><a href="right-sidebar.html">History</a></li>
                            <li><a href="no-sidebar.html">Profile</a></li>
                        </ul>
                    </nav>
                </div>
            </header>
        </div>

        <!-- Main Content -->
        <div id="main-wrapper">
            <div class="wrapper style1">
                <div class="inner">
                    <div class="link-container">
                        <h2>Link Your Electric Vehicle</h2>
                        <p>Connect your EV to start charging at Cephra stations</p>

                        <img src="images/ConnectCar.gif" alt="Connect Car" style="max-width: 100%; height: auto; margin: 20px 0;">

                        <form method="post" id="linkForm">
                            <div class="terms-checkbox">
                                <input type="checkbox" id="terms" name="terms" required>
                                <label for="terms">By linking, I agree to the <a href="#" onclick="showTerms(); return false;">Terms & Conditions</a></label>
                            </div>

                            <?php if (isset($error)): ?>
                                <div class="error-message"><?php echo htmlspecialchars($error); ?></div>
                            <?php endif; ?>

                            <button type="submit" name="link_car" class="link-button" id="linkBtn" disabled>
                                Link My Car
                            </button>
                        </form>
                    </div>
                </div>
            </div>
        </div>

        <!-- Navigation Buttons -->
        <div class="nav-buttons">
            <button class="nav-button" onclick="window.location.href='dashboard.php'" title="Home">🏠</button>
            <button class="nav-button" onclick="window.location.href='ChargingPage.php'" title="Charge">🔋</button>
            <button class="nav-button" onclick="window.location.href='dashboard.php'" title="Profile">👤</button>
        </div>
    </div>

    <!-- Scripts -->
    <script src="assets/js/jquery.min.js"></script>
    <script src="assets/js/jquery.dropotron.min.js"></script>
    <script src="assets/js/browser.min.js"></script>
    <script src="assets/js/breakpoints.min.js"></script>
    <script src="assets/js/util.js"></script>
    <script src="assets/js/main.js"></script>

    <script>
        function showTerms() {
            const termsText = `CEPHRA EV LINKING TERMS AND CONDITIONS
Effective Date: <?php echo date('Y-m-d'); ?>
Version: 1.0

1. ACCEPTANCE OF TERMS
By linking your electric vehicle (EV) to the Cephra app (the "Service"), you agree to these Terms.

2. LINKING PURPOSE
Linking enables ticketing, charging session history, and status updates within the app.

3. DATA COLLECTED
Vehicle identifiers, session timestamps, kWh consumed, payment references, and diagnostic status necessary for Service delivery.

4. USER RESPONSIBILITIES
You confirm you are authorized to link the vehicle and will keep your account secure.

5. CONSENT TO COMMUNICATIONS
You consent to in-app notifications and transactional emails about charging and tickets.

6. PRIVACY
We process data per our Privacy Policy. Data is retained only as long as needed for the Service.

7. LIMITATIONS
The Service provides information "as is" and availability may vary by station and network conditions.

8. SECURITY
We employ reasonable safeguards, but you acknowledge inherent risks in networked systems.

9. UNLINKING
You may unlink your vehicle at any time from the app; some records may be retained for compliance.

10. LIABILITY
To the maximum extent permitted by law, Cephra is not liable for indirect or consequential damages.

11. GOVERNING LAW
These Terms are governed by the laws of Pasay City, Philippines.

12. CONTACT
Cephra Support — support@cephra.com | +63 2 8XXX XXXX

By checking "I agree", you confirm you have read and accept these Terms.`;

            // Create modal dialog
            const modal = document.createElement('div');
            modal.style.cssText = `
                position: fixed;
                top: 0;
                left: 0;
                width: 100%;
                height: 100%;
                background: rgba(0,0,0,0.8);
                display: flex;
                align-items: center;
                justify-content: center;
                z-index: 10000;
            `;

            const modalContent = document.createElement('div');
            modalContent.style.cssText = `
                background: white;
                padding: 20px;
                border-radius: 10px;
                max-width: 600px;
                max-height: 80vh;
                overflow-y: auto;
                position: relative;
            `;

            modalContent.innerHTML = `
                <h3>Terms and Conditions</h3>
                <pre style="white-space: pre-wrap; font-family: inherit; font-size: 14px; line-height: 1.5;">${termsText}</pre>
                <button onclick="this.closest('div').parentElement.remove()" style="margin-top: 20px; padding: 10px 20px; background: #007bff; color: white; border: none; border-radius: 5px; cursor: pointer;">OK</button>
            `;

            modal.appendChild(modalContent);
            document.body.appendChild(modal);
        }

        // Form validation
        document.getElementById('linkForm').addEventListener('submit', function(e) {
            const termsCheckbox = document.getElementById('terms');
            if (!termsCheckbox.checked) {
                e.preventDefault();
                alert('Please agree to the Terms & Conditions to continue.');
                return false;
            }
        });

        // Enable/disable button based on terms checkbox
        document.getElementById('terms').addEventListener('change', function() {
            const linkBtn = document.getElementById('linkBtn');
            if (this.checked) {
                linkBtn.disabled = false;
                linkBtn.style.background = '#4CAF50';
                linkBtn.style.cursor = 'pointer';
            } else {
                linkBtn.disabled = true;
                linkBtn.style.background = '#cccccc';
                linkBtn.style.cursor = 'not-allowed';
            }
        });

        // Initialize button state on page load
        document.addEventListener('DOMContentLoaded', function() {
            const termsCheckbox = document.getElementById('terms');
            const linkBtn = document.getElementById('linkBtn');
            if (!termsCheckbox.checked) {
                linkBtn.disabled = true;
                linkBtn.style.background = '#cccccc';
                linkBtn.style.cursor = 'not-allowed';
            }
        });
    </script>
</body>
</html>

File: src/main/java/cephra/Phone/link.java

package cephra.Phone;

import javax.swing.SwingUtilities;

public class link extends javax.swing.JPanel {

    public link() {
        initComponents();
        setPreferredSize(new java.awt.Dimension(350, 750));
        setSize(350, 750);
        setupLabelPosition(); // Set label position
        
        // Check if car is already linked and user has a battery level
        if (cephra.Phone.AppState.isCarLinked) {
            try {
                String username = cephra.CephraDB.getCurrentUsername();
                if (username != null && !username.isEmpty()) {
                    int batteryLevel = cephra.CephraDB.getUserBatteryLevel(username);
                    if (batteryLevel != -1) {
                        // Car is linked and battery is initialized - go to Porsche
                        javax.swing.SwingUtilities.invokeLater(new Runnable() {
                            public void run() {
                                java.awt.Window[] windows = java.awt.Window.getWindows();
                                for (java.awt.Window window : windows) {
                                    if (window instanceof cephra.Frame.Phone) {
                                        cephra.Frame.Phone phoneFrame = (cephra.Frame.Phone) window;
                                        phoneFrame.switchPanel(new cephra.Phone.PorscheTaycan());
                                        break;
                                    }
                                }
                            }
                        });
                    }
                }
            } catch (Exception e) {
                System.err.println("Error checking battery level in link constructor: " + e.getMessage());
            }
        }
    }

    
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        Linkcar = new javax.swing.JButton();
        charge = new javax.swing.JButton();
        historybutton = new javax.swing.JButton();
        profilebutton = new javax.swing.JButton();
        homebutton = new javax.swing.JButton();
        termscondition = new javax.swing.JCheckBox();
        jLabel1 = new javax.swing.JLabel();

        setMaximumSize(new java.awt.Dimension(350, 750));
        setPreferredSize(new java.awt.Dimension(350, 750));
        setLayout(null);

        Linkcar.setBorder(null);
        Linkcar.setBorderPainted(false);
        Linkcar.setContentAreaFilled(false);
        Linkcar.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                LinkcarActionPerformed(evt);
            }
        });
        add(Linkcar);
        Linkcar.setBounds(50, 560, 240, 50);

        charge.setBorder(null);
        charge.setBorderPainted(false);
        charge.setContentAreaFilled(false);
        charge.setFocusPainted(false);
        charge.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                chargeActionPerformed(evt);
            }
        });
        add(charge);
        charge.setBounds(40, 670, 50, 60);

        historybutton.setBorder(null);
        historybutton.setBorderPainted(false);
        historybutton.setContentAreaFilled(false);
        historybutton.setFocusPainted(false);
        historybutton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                historybuttonActionPerformed(evt);
            }
        });
        add(historybutton);
        historybutton.setBounds(230, 653, 50, 100);

        profilebutton.setBorder(null);
        profilebutton.setBorderPainted(false);
        profilebutton.setContentAreaFilled(false);
        profilebutton.setFocusPainted(false);
        profilebutton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                profilebuttonActionPerformed(evt);
            }
        });
        add(profilebutton);
        profilebutton.setBounds(290, 660, 80, 70);

        homebutton.setBorder(null);
        homebutton.setBorderPainted(false);
        homebutton.setContentAreaFilled(false);
        homebutton.setFocusPainted(false);
        homebutton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                homebuttonActionPerformed(evt);
            }
        });
        add(homebutton);
        homebutton.setBounds(150, 680, 40, 40);

        termscondition.setBackground(new java.awt.Color(255, 255, 255));
        termscondition.setText("By linking, I agree to the Terms & Conditions ");
        termscondition.setBorder(null);
        termscondition.setContentAreaFilled(false);
        termscondition.setFocusPainted(false);
        termscondition.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                termsconditionActionPerformed(evt);
            }
        });
        add(termscondition);
        termscondition.setBounds(40, 620, 310, 30);

        jLabel1.setIcon(new javax.swing.ImageIcon(getClass().getResource("/cephra/Cephra Images/ConnectCar.gif"))); // NOI18N
        add(jLabel1);
        jLabel1.setBounds(-15, 0, 398, 750);
    }// </editor-fold>//GEN-END:initComponents

    // CUSTOM CODE - DO NOT REMOVE - NetBeans will regenerate form code but this method should be preserved
    // Setup label position to prevent NetBeans from changing it
    private void setupLabelPosition() {
        if (jLabel1 != null) {
            jLabel1.setBounds(-15, 0, 398, 750);
        }
    }

    private void chargeActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_chargeActionPerformed
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                java.awt.Window[] windows = java.awt.Window.getWindows();
                for (java.awt.Window window : windows) {
                    if (window instanceof cephra.Frame.Phone) {
                        cephra.Frame.Phone phoneFrame = (cephra.Frame.Phone) window;
                        phoneFrame.switchPanel(new cephra.Phone.serviceoffered());
                        break;
                    }
                }
            }
        });
    }//GEN-LAST:event_chargeActionPerformed

    private void historybuttonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_historybuttonActionPerformed
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                java.awt.Window[] windows = java.awt.Window.getWindows();
                for (java.awt.Window window : windows) {
                    if (window instanceof cephra.Frame.Phone) {
                        cephra.Frame.Phone phoneFrame = (cephra.Frame.Phone) window;
                        phoneFrame.switchPanel(new cephra.Phone.phonehistory());
                        break;
                    }
                }
            }
        });
    }//GEN-LAST:event_historybuttonActionPerformed

    private void profilebuttonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_profilebuttonActionPerformed
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                java.awt.Window[] windows = java.awt.Window.getWindows();
                for (java.awt.Window window : windows) {
                    if (window instanceof cephra.Frame.Phone) {
                        cephra.Frame.Phone phoneFrame = (cephra.Frame.Phone) window;
                        phoneFrame.switchPanel(new cephra.Phone.Profile());
                        break;
                    }
                }
            }
        });
    }//GEN-LAST:event_profilebuttonActionPerformed

    private void homebuttonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_homebuttonActionPerformed
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                java.awt.Window[] windows = java.awt.Window.getWindows();
                for (java.awt.Window window : windows) {
                    if (window instanceof cephra.Frame.Phone) {
                        cephra.Frame.Phone phoneFrame = (cephra.Frame.Phone) window;
                        phoneFrame.switchPanel(new cephra.Phone.home());
                        break;
                    }
                }
            }
        });
    }//GEN-LAST:event_homebuttonActionPerformed

    private void LinkcarActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_LinkcarActionPerformed
          SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                // Generate and set battery level for current user when linking car
                String username = cephra.CephraDB.getCurrentUsername();
                if (username != null && !username.isEmpty()) {
                            // Initialize battery to random level (15-50%) only if not already initialized
        int currentBattery = cephra.CephraDB.getUserBatteryLevel(username);
        if (currentBattery == -1) {
            // No battery initialized yet - create new random battery
            java.util.Random random = new java.util.Random();
            int batteryLevel = 15 + random.nextInt(36); // 15 to 50
            cephra.CephraDB.setUserBatteryLevel(username, batteryLevel);
            System.out.println("Link: Initialized battery level for " + username + " to " + batteryLevel + "% when linking car");
                } else {
            // Battery already exists - keep the current level
            System.out.println("Link: Battery level for " + username + " already exists: " + currentBattery + "% - keeping current level");
        }
                    
                    // Ensure database update is complete before switching panels
                    try {
                        Thread.sleep(100); // Small delay to ensure database update is complete
                    } catch (InterruptedException e) {
                        Thread.currentThread().interrupt();
                    }
                }
                
                cephra.Phone.AppState.isCarLinked = true;
                java.awt.Window[] windows = java.awt.Window.getWindows();
                for (java.awt.Window window : windows) {
                    if (window instanceof cephra.Frame.Phone) {
                        cephra.Frame.Phone phoneFrame = (cephra.Frame.Phone) window;
                        // Go directly to PorscheTaycan instead of serviceoffered
                        phoneFrame.switchPanel(new cephra.Phone.PorscheTaycan());
                        break;
                    }
                }
            }
        });
    }//GEN-LAST:event_LinkcarActionPerformed

    private void termsconditionActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_termsconditionActionPerformed
        if (termscondition.isSelected()) {
            // Show Terms and Conditions in a dialog
            showTermsAndConditions();
        }
    }//GEN-LAST:event_termsconditionActionPerformed


    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton Linkcar;
    private javax.swing.JButton charge;
    private javax.swing.JButton historybutton;
    private javax.swing.JButton homebutton;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JButton profilebutton;
    private javax.swing.JCheckBox termscondition;
    // End of variables declaration//GEN-END:variables

    @Override
    public void addNotify() {
        super.addNotify();
        javax.swing.SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                setupLabelPosition();
            }
        });
    }

    private void showTermsAndConditions() {
        String termsText = getLinkTermsAndConditionsText();

        String safeText = termsText
                .replace("&", "&amp;")
                .replace("<", "&lt;")
                .replace(">", "&gt;");
        String html = "<html><head><style>" +
                "body{font-family:'Segoe UI',sans-serif;"
                + "font-size:12px;"
                + "line-height:1.5;"
                + "text-align:justify;margin:0;}" +
                ".container{padding:0 4px;}" +
                "</style></head><body><div class='container'>" +
                safeText.replace("\n", "<br/>") +
                "</div></body></html>";

        javax.swing.JEditorPane editorPane = new javax.swing.JEditorPane("text/html", html);
        editorPane.setEditable(false);
        editorPane.setOpaque(true);
        editorPane.setFocusable(false);
        editorPane.setHighlighter(null);
        editorPane.setCursor(java.awt.Cursor.getPredefinedCursor(java.awt.Cursor.DEFAULT_CURSOR));
        if (editorPane.getCaret() != null) {
            editorPane.getCaret().setVisible(false);
            editorPane.getCaret().setSelectionVisible(false);
        }

        javax.swing.JScrollPane scrollPane = new javax.swing.JScrollPane(editorPane);
        scrollPane.setPreferredSize(new java.awt.Dimension(600, 400));
        scrollPane.setBorder(null);
        scrollPane.setWheelScrollingEnabled(true);
        scrollPane.setHorizontalScrollBarPolicy(javax.swing.ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
        scrollPane.setVerticalScrollBarPolicy(javax.swing.ScrollPaneConstants.VERTICAL_SCROLLBAR_NEVER);

        java.awt.Window owner = javax.swing.SwingUtilities.getWindowAncestor(this);
        final javax.swing.JDialog dialog = new javax.swing.JDialog(owner, "Terms and Conditions", java.awt.Dialog.ModalityType.APPLICATION_MODAL);
        dialog.setUndecorated(true);
        dialog.setDefaultCloseOperation(javax.swing.JDialog.DO_NOTHING_ON_CLOSE);

        javax.swing.JPanel content = new javax.swing.JPanel(new java.awt.BorderLayout());
        content.setBorder(javax.swing.BorderFactory.createEmptyBorder(12, 12, 12, 12));
        content.add(scrollPane, java.awt.BorderLayout.CENTER);

        javax.swing.JButton ok = new javax.swing.JButton("OK");
        ok.setFocusPainted(false);
        ok.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                dialog.dispose();
            }
        });
        javax.swing.JPanel buttons = new javax.swing.JPanel();
        buttons.setBackground(java.awt.Color.WHITE); 
        buttons.add(ok);
        content.add(buttons, java.awt.BorderLayout.SOUTH);

        dialog.setContentPane(content);
        dialog.getContentPane().setBackground(java.awt.Color.WHITE); 
        dialog.setBackground(java.awt.Color.WHITE);
        dialog.setSize(320, 600);
        if (owner != null) {
            dialog.setLocationRelativeTo(owner);
            java.awt.Point current = dialog.getLocation();
            dialog.setLocation(current.x - 3, current.y);
        }
        try {
            editorPane.setCaretPosition(0);
        } catch (Exception ignore) {}
        javax.swing.SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                try { editorPane.setCaretPosition(0); } catch (Exception ignore) {}
                javax.swing.JViewport vp = scrollPane.getViewport();
                if (vp != null) {
                    vp.setViewPosition(new java.awt.Point(0, 0));
                }
            }
        });
        dialog.setVisible(true);
    }

    private String getLinkTermsAndConditionsText() {
        return "CEPHRA EV LINKING TERMS AND CONDITIONS\n" +
               "Effective Date: " + java.time.LocalDate.now() + "\n" +
               "Version: 1.0\n\n" +
               "1. ACCEPTANCE OF TERMS\n" +
               "By linking your electric vehicle (EV) to the Cephra app (the \"Service\"), you agree to these Terms.\n\n" +
               "2. LINKING PURPOSE\n" +
               "Linking enables ticketing, charging session history, and status updates within the app.\n\n" +
               "3. DATA COLLECTED\n" +
               "Vehicle identifiers, session timestamps, kWh consumed, payment references, and diagnostic status necessary for Service delivery.\n\n" +
               "4. USER RESPONSIBILITIES\n" +
               "You confirm you are authorized to link the vehicle and will keep your account secure.\n\n" +
               "5. CONSENT TO COMMUNICATIONS\n" +
               "You consent to in-app notifications and transactional emails about charging and tickets.\n\n" +
               "6. PRIVACY\n" +
               "We process data per our Privacy Policy. Data is retained only as long as needed for the Service.\n\n" +
               "7. LIMITATIONS\n" +
               "The Service provides information \"as is\" and availability may vary by station and network conditions.\n\n" +
               "8. SECURITY\n" +
               "We employ reasonable safeguards, but you acknowledge inherent risks in networked systems.\n\n" +
               "9. UNLINKING\n" +
               "You may unlink your vehicle at any time from the app; some records may be retained for compliance.\n\n" +
               "10. LIABILITY\n" +
               "To the maximum extent permitted by law, Cephra is not liable for indirect or consequential damages.\n\n" +
               "11. GOVERNING LAW\n" +
               "These Terms are governed by the laws of Pasay City, Philippines.\n\n" +
               "12. CONTACT\n" +
               "Cephra Support — support@cephra.com | +63 2 8XXX XXXX\n\n" +
               "By checking \"I agree\", you confirm you have read and accept these Terms.";
    }
}
